src/app/Fake.DotNet.Testing.Expecto/Expecto.fs

```diff
@@ -0,0 +1,134 @@
+        let appendIfNotEqual other value s (sb: StringBuilder.StringBuilder) =
+            if other = value
+            then sprintf "%s%A " s value |> sb.Append
+            else sb
+        let appendList list s (sb: StringBuilder.StringBuilder) =
+            let filtered = list |> List.filter (String.IsNullOrWhiteSpace >> not)
+            if List.isEmpty filtered then sb
+            else
+                filtered |> String.separated " " |> sprintf "%s%s " s |> sb.Append
+        StringBuilder.StringBuilder()
+        |> appendIfTrue this.Debug "--debug "
+        |> appendIfTrue this.Parallel "--parallel "
+        |> appendIfNotEqual 0 this.ParallelWorkers "--parallel-workers "
+        |> appendIfTrue this.FailOnFocusedTests "--fail-on-focused-tests "
+        |> appendIfTrue this.Summary "--summary "
+        |> appendIfTrue this.SummaryLocation "--summary-location "
+        |> appendIfTrue (not this.Parallel) "--sequenced "
+        |> appendIfTrue this.PrintVersion "--version "
+        |> appendIfTrue this.ListTests "--list-tests "
+        |> appendIfNotNullOrWhiteSpace this.Filter "--filter "
+        |> appendIfNotNullOrWhiteSpace this.FilterTestCase "--filter-test-case "
+        |> appendIfNotNullOrWhiteSpace this.FilterTestList "--filter-test-list "
+        |> appendList this.Run "--run "
+        |> appendList this.CustomArgs ""
+        |> StringBuilder.toText
+
+    static member DefaultParams =
+        {
+            Debug = false
+            Parallel = true
+            ParallelWorkers = 0
+            Filter = ""
+            FilterTestCase = ""
+            FailOnFocusedTests = true
+            FilterTestList = ""
+            PrintVersion = true
+            Run = []
+            ListTests = false
+            // Summary = true somehow breakes Expecto TeamCity printer
+            Summary = false
+            SummaryLocation = false
+            CustomArgs = []
+            WorkingDirectory = ""
+        }
+
+let run (setParams : Params -> Params) (assemblies : string seq) =
+    let details = assemblies |> String.separated ", "
+    use __ = Trace.traceTask "Expecto" details
+
+    let runAssembly testAssembly =
+        let exitCode =
+            let fakeStartInfo testAssembly args  =
+                let workingDir =
+                    if String.isNotNullOrEmpty args.WorkingDirectory
+                    then args.WorkingDirectory else Fake.IO.Path.getDirectory testAssembly
+                (fun (info: ProcStartInfo) ->
+                    { info with 
+                        FileName = testAssembly
+                        Arguments = string args
+                        WorkingDirectory = workingDir } 
+                    // Pass environment variables to the expecto console process in order to let it detect it's running on TeamCity
+                    // (it checks TEAMCITY_PROJECT_NAME <> null specifically).
+                    |> Process.setEnvironmentVariables (Environment.environVars()) )
```

Comment:
It should be working by just deleting this like. AFAICS you don't add any special environment variables or remove them, correct?

Reply:
I just copied over this from the legacy code based on the comment. Removed.